test/new-e2e/shared-components/hostname_ec2_test.go

```diff
@@ -0,0 +1,100 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package testinfradefinition
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/e2e/client"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+)
+
+type agentSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+}
+
+func TestAgentHostnameEC2Suite(t *testing.T) {
+	e2e.Run(t, &agentSuite{}, e2e.AgentStackDef(nil))
+}
+
+// https://github.com/DataDog/datadog-agent/blob/main/pkg/util/hostname/README.md#the-current-logic
+func (v *agentSuite) TestAgentHostnameDefaultsToResourceId() {
+	v.UpdateEnv(e2e.AgentStackDef(nil, agentparams.WithAgentConfig("")))
+
+	metadata := client.NewEC2Metadata(v.Env().VM)
+	err := v.Env().Agent.WaitForReady()
+	assert.NoError(v.T(), err)
+
+	hostname := v.Env().Agent.Hostname()
+
+	// Default configuration of hostname for EC2 instances is the resource-id
+	resourceId := metadata.Get("instance-id")
+	assert.Contains(v.T(), hostname, resourceId)
+}
+
+func (v *agentSuite) TestAgentConfigHostnameVarOverride() {
+	v.UpdateEnv(e2e.AgentStackDef(nil, agentparams.WithAgentConfig("hostname: hostname.from.var")))
+	err := v.Env().Agent.WaitForReady()
+	assert.NoError(v.T(), err)
+
+	hostname := v.Env().Agent.Hostname()
+	assert.Contains(v.T(), hostname, "hostname.from.var")
+}
+
+func (v *agentSuite) TestAgentConfigHostnameFileOverride() {
+	fileContent := "hostname.from.file"
+	v.Env().VM.Execute(fmt.Sprintf(`echo "%s" | tee /tmp/hostname`, fileContent))
```

Comment:
I can bring this up as a request at the next squad office hours, there is a [CopyInlineFile](https://github.com/DataDog/test-infra-definitions/blob/main/components/command/unixOSCommand.go#L44-L57) which does this but isn't exposed as part of the e2e client interface

this would be nice to wrap similarly as it would include the pulumi resource teardown between tests 👍